MDANSE/Src/MDANSE/Framework/Jobs/DynamicCoherentStructureFactor.py

```diff
@@ -324,19 +331,24 @@ def run_step(self, index: int) -> tuple[int, dict[str, np.ndarray] | None]:
                 f"The unit cell is VARIABLE with the standard deviation of {self._cell_std}. This analysis should not be used with NPT runs! PLEASE CHECK YOUR RESULTS CAREFULLY."
             )
 
-        return index, rho
+        return index, (rho, np.sum(qvec_weights))
 
     def combine(self, index: int, x: np.ndarray):
         """Add partial results to the final array."""
         if x is not None:
+            norm = x[1]
+            rho = x[0]
             n_configs = self.configuration["frames"]["n_configs"]
             for pair_str, (label_i, label_j) in self.labels:
                 # F_ab(Q,t) = F_ba(Q,t) this is valid as long as
                 # n_configs is sufficiently large
-                corr = correlate(x[label_i], x[label_j][:n_configs], mode="valid").T[
-                    0
-                ] / (n_configs * x[label_i].shape[1])
+                corr = correlate(
+                    rho[label_i], rho[label_j][:n_configs], mode="valid"
+                ).T[0] / (n_configs * norm)
                 self._outputData[f"dcsf/f(q,t)/{pair_str}"][index, :] += corr.real
+        else:
+            for pair_str, (_, _) in self.labels:
```

Comment:
```suggestion
            for pair_str, _ in self.labels:
```
Don't really need to unpack if they're both ignored.